internal/proto/hotstuffpb/convert_test.go

```diff
@@ -78,6 +78,8 @@ func TestConvertBlock(t *testing.T) {
 }
 
 // TODO(meling): disabled since it fails after the package-restructuring and I haven't been able to fix it yet.
+// TODO(AlanRostem): discovered the bug - the metadata per replica is nil and is needed by BLS12 to generate public keys.
```

Comment:
Great! On that note, [Gorums v0.9.0](https://github.com/relab/gorums/releases/tag/v0.9.0) now supports passing [metadata per message](https://github.com/relab/gorums/blob/ad120bcf1ed7628fe12f9b083894f38f8992259f/tests/metadata/metadata_test.go#L101) as well, not only per connection.